compiler/src/iree/compiler/Dialect/Stream/Analysis/ResourceUsage.cpp

```diff
@@ -548,6 +561,19 @@ class ValueResourceUsage : public AbstractResourceUsage<DFX::ValueElement> {
                 *this, value, DFX::Resolution::REQUIRED);
             getState() ^= valueUsage.getState();
           }
+
```

Comment:
nesting the cases within a case feels a bit odd - are there any other ops that scf.yield could be parented in? we should probably default to failing or something if if/while/for are not matched as otherwise it'd cause hard to debug issues

Reply:
Do we have a mechanism for logging errors here? In theory the only issue is additional control flow analysis which would already fail if we encountered it.